File: 20.js
function timesDescending(top) {
  var i = top;
  var sum = 1;
  while (i <= top && i > 0) {
    sum = sum * i;
    console.log(i);
    i--;
  }
  console.log(sum);
  return sum; 
}

function addSumOfDigits(num) {
  var arr = num.toString()
        .replace(/e.*$/,'')
        .replace('.','')
        .split('');
  console.log('arr', arr);
  return arr.reduce(function(previousValue, currentValue){
    return parseFloat(previousValue, 10) + parseFloat(currentValue, 10);
  });
}

console.log(addSumOfDigits(timesDescending(100)));

File: 20.fs
let timesDescending top =
    let mutable i = top : int
    let mutable sum = (float 1.0)
    while i <= top && i > 0 do
      sum <- sum * (float i)
      i <- i - 1
    sum

let addSumOfDigits num =
    let arr = num.ToString().Replace(".","").ToCharArray()
    Console.Write(arr)
    List.reduce (fun prev curr -> prev + curr) arr

printfn "%f" (timesDescending (100))
printfn "addSumOfDigits %A " (addSumOfDigits (timesDescending 100))